src/main/java/uk/gov/hmcts/reform/judicialapi/service/impl/JudicialUserServiceImpl.java

```diff
@@ -49,4 +75,99 @@
                 .body(ormResponses);
     }
 
+    /**
+     * Returns the User Profile for Refresh Assignments.
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public ResponseEntity<Object> fetchUserProfileByServiceNames(String ccdServiceNames, PageRequest pageRequest) {
```

Comment:
Looks very big method try to modularise code by delegating task to smaller methods.